### PR TITLE
DAOS-13548 test: Add a test for pil4dfs under fi.

### PR DESCRIPTION
**Pull request by ashleypittman**
- DAOS-13548 test: Add a test for pil4dfs under fi.
- Expand test.

### CODE REVIEW

**Comment by ashleypittman** [comment]
This PR adds but leaves disabled a new test for pil4dfs running under the interception library, the results of the new test can be seen at https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12259/2/

Many of the issues relate to logging and will be fixed by https://github.com/daos-stack/daos/pull/12218 however others relate to memory leaks, incorrect error handling and use-after-free of container handles.  @wiliamhuang can you enable this test and resolve the issues once 12218 has landed.

**Comment by wiliamhuang** [comment]
> This PR adds but leaves disabled a new test for pil4dfs running under the interception library, the results of the new test can be seen at https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12259/2/
> Many of the issues relate to logging and will be fixed by #12218 however others relate to memory leaks, incorrect error handling and use-after-free of container handles. @wiliamhuang can you enable this test and resolve the issues once 12218 has landed.

ok. I will check it out once https://github.com/daos-stack/daos/pull/12218 is landed. Thank you!

**Comment by ashleypittman** [comment]
> ok. I will check it out once #12218 is landed. Thank you!

This PR can be reviewed/landed etc whilst #12218 is still being worked so can you review the test in parallel with progressing that.  Then you can create a future PR to enable the test and resolve the issues.

**Review by daltonbohning — APPROVED** [review]

**Review by wiliamhuang — APPROVED** [review]

File: utils/node_local_test.py [hunk 1]
```diff
@@ -4989,27 +4993,27 @@ def _explain():
...
-                                   show_memleaks=show_memleaks,
-                                   quiet=True,
-                                   skip_fi=True,
-                                   leak_wf=wf)
+            self._fi_loc = log_test(self._aft.conf,
+                                    self.log_file,
+                                    show_memleaks=show_memleaks,
+                                    quiet=True,
+                                    skip_fi=True,
+                                    leak_wf=wf)
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply utils/node_local_test.py:5004]: is there a way to tell pylint to expect that style instead?

[the remark this answers is not in the document]
Comment (daltonbohning) [reply utils/node_local_test.py:5004]: Maybe? But we have a bit of code following this style. And I think @ashleypittman prefers this style generally

**Pull request merged**